### PR TITLE
fix(server): cannot render email template

### LINKED ISSUE #13955: Setting up email notifications causes server error
### The bug

Im trying to set up email notifications for my immich server. After filling in the settings and trying to send a test email, I get a server error. 

I was setting it up with google smtp server following [this guide](https://immich.app/docs/guides/smtp-gmail).

### The OS that Immich Server is running on

Arch Linux

### Version of Immich Server

v1.120.0

### Version of Immich Mobile App

N/A

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-tailscale:
    image: tailscale/tailscale:latest
    container_name: immich-tailscale
    cap_add:
      - net_admin
      - sys_module
    volumes:
      - ./tailscale:/var/lib/tailscale
      - ./tailscale-config.json:/tailscale-config.json
#      - /lib/modules:/lib/modules:ro
    devices:
      - /dev/net/tun
    env_file:
      - ./.env
    environment:
      - TS_EXTRA_ARGS=--advertise-tags=tag:container --reset
      - TS_STATE_DIR=/var/lib/tailscale
      - TS_SERVE_CONFIG=/tailscale-config.json
      - TS_USERSPACE=false
    hostname: immich
    restart: always

  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    extends:
       file: hwaccel.transcoding.yml
       service: vaapi # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    volumes:
      # Do not edit the next line. If you want to change the media storage location on your system, edit the value of UPLOAD_LOCATION in the .env file
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    # ports:
      # - 2283:2283
    depends_on:
      - redis
      - database
      - immich-tailscale
    network_mode: service:immich-tailscale
    restart: always
    healthcheck:
      disable: false

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
#    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
#       file: hwaccel.ml.yml
#       service: cuda # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    # network_mode: service:immich-tailscale
    restart: always
    healthcheck:
      disable: false

  redis:
    container_name: immich_redis
    image: docker.io/redis:6.2-alpine@sha256:2d1463258f2764328496376f5d965f20c6a67f66ea2b06dc42af351f75248792
    healthcheck:
      test: redis-cli ping || exit 1
    # network_mode: service:immich-tailscale
    restart: always

  database:
    container_name: immich_postgres
    image: docker.io/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
      POSTGRES_INITDB_ARGS: '--data-checksums'
    volumes:
      # Do not edit the next line. If you want to change the database storage location on your system, edit the value of DB_DATA_LOCATION in the .env file
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' || exit 1; Chksum="$$(psql --dbname='${DB_DATABASE_NAME}' --username='${DB_USERNAME}' --tuples-only --no-align --command='SELECT COALESCE(SUM(checksum_failures), 0) FROM pg_stat_database')"; echo "checksum failure count is $$Chksum"; [ "$$Chksum" = '0' ] || exit 1
      interval: 5m
      start_interval: 30s
      start_period: 5m
    command: ["postgres", "-c", "shared_preload_libraries=vectors.so", "-c", 'search_path="$$user", public, vectors', "-c", "logging_collector=on", "-c", "max_wal_size=2GB", "-c", "shared_buffers=512MB", "-c", "wal_compression=on"]
    # network_mode: service:immich-tailscale
    restart: always

volumes:
  model-cache:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./library
# The location where your database files are stored
DB_DATA_LOCATION=./postgres

# To set a timezone, uncomment the next line and change Etc/UTC to a TZ identifier from this list: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones#List
TZ=<redacted>

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
# Please use only the characters `A-Za-z0-9`, without special characters or spaces
DB_PASSWORD=<redacted>

# The values below this line do not need to be changed
###################################################################################
DB_USERNAME=<redacted>
DB_DATABASE_NAME=<redacted>
```

### Reproduction steps

1. Set up a new docker server.
2. Fill in email notification details (with freshly created google app password).
3. Hit send test email and save.
...

### Relevant log output

```shell
[Nest] 16  - 11/06/2024, 6:42:14 PM   ERROR [Api:ErrorInterceptor~yeh8f2x7] Unknown error: Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.
Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.
    at Z (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:79:108)
    at Yc (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:81:98)
    at $c (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:80:140)
    at Z (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:345)
    at Xc (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:72:481)
    at Z (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)
    at Yc (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:81:98)
    at Xc (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:73:145)
    at Z (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:78:89)
    at Xc (/usr/src/app/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:75:509)
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alextran1502**
Fixes #13955

### CODE REVIEW

**Comment by bo0tzz** [comment]
Should we add a test that renders the template?

**Comment by alextran1502** [comment]
> Should we add a test that renders the template?

@bo0tzz Done

File: server/src/repositories/notification.repository.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    } as unknown as Mocked<ILoggerRepository>;
+
+    sut = new NotificationRepository(loggerMock);
+  });
+
+  describe('renderEmail', () => {
+    it('should render the email correctly for TEST_EMAIL template', async () => {
+      const request: EmailRenderRequest = {
+        template: EmailTemplate.TEST_EMAIL,
+        data: { displayName: 'Alen Turing', baseUrl: 'http://localhost' },
```
Comment (bo0tzz) [inline server/src/repositories/notification.repository.spec.ts:23]:
```suggestion
        data: { displayName: 'Alan Turing', baseUrl: 'http://localhost' },
```

**Review by danieldietzler — APPROVED** [review]
It's Alan 🤫

File: server/src/repositories/notification.repository.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      const result = await sut.renderEmail(request);
+
+      expect(result.html).toContain('<!DOCTYPE html PUBLIC');
+      expect(result.text).toContain('test email');
+    });
+
+    it('should render the email correctly for WELCOME template', async () => {
+      const request: EmailRenderRequest = {
+        template: EmailTemplate.WELCOME,
+        data: { displayName: 'Alen Turing', username: 'turing', baseUrl: 'http://localhost' },
```
Comment (bo0tzz) [inline server/src/repositories/notification.repository.spec.ts:35]:
```suggestion
        data: { displayName: 'Alan Turing', username: 'turing', baseUrl: 'http://localhost' },
```

**Comment by bo0tzz** [comment]
rip lol

**Pull request merged**